test/extensions/access_loggers/grpc/tcp_grpc_access_log_integration_test.cc

```diff
@@ -786,5 +795,186 @@ TEST_P(TcpGrpcAccessLogIntegrationTest, SslNotTerminatedWithJA3NoSNI) {
+  ssl_options.setClientEcdsaCert(false);
+  ssl_options.setCipherSuites({"ECDHE-RSA-AES128-GCM-SHA256"});
+  ssl_options.setTlsVersion(envoy::extensions::transport_sockets::tls::v3::TlsParameters::TLSv1_2);
+  setupSslConnection(/*expect_connection_open=*/false,
+                     /*ssl_options=*/ssl_options, /*curves_list=*/"P-256");
+
+  ASSERT_TRUE(waitForAccessLogConnection());
+  ASSERT_TRUE(waitForAccessLogStream());
+  ASSERT_TRUE(
+      waitForAccessLogRequest(fmt::format(R"EOF(
+identifier:
+  node:
+    id: node_name
+    cluster: cluster_name
+    locality:
+      zone: zone_name
+    user_agent_name: "envoy"
+  log_name: foo
+tcp_logs:
+  log_entry:
+    common_properties:
+      downstream_remote_address:
+        socket_address:
+          address: {0}
+      downstream_local_address:
+        socket_address:
+          address: {0}
+      downstream_transport_failure_reason: "TLS_error:|268435581:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED:TLS_error_end:TLS_error_end"
+      access_log_type: NotSet
+      downstream_direct_remote_address:
+        socket_address:
+          address: {0}
+      tls_properties:
+        tls_version: TLSv1_2
+        tls_cipher_suite:
+          value: 49199
+        local_certificate_properties:
+          subject_alt_name:
+            - uri: "spiffe://lyft.com/backend-team"
+          subject: "emailAddress=backend-team@lyft.com,CN=Test Backend Team,OU=Lyft Engineering,O=Lyft,L=San Francisco,ST=California,C=US"
+        peer_certificate_properties:
+          subject_alt_name:
+            - uri: "spiffe://lyft.com/frontend-team"
+          subject: "emailAddress=frontend-team@lyft.com,CN=Test Frontend Team,OU=Lyft Engineering,O=Lyft,L=San Francisco,ST=California,C=US"
+          issuer: "CN=Test CA,OU=Lyft Engineering,O=Lyft,L=San Francisco,ST=California,C=US"
+      upstream_remote_address:
+        socket_address: {{}}
+      upstream_local_address:
+        socket_address: {{}}
+    connection_properties: {{}}
+)EOF",
+                                          Network::Test::getLoopbackAddressString(ipVersion()))));
+  cleanup();
+}
+
+TEST_P(TcpGrpcAccessLogIntegrationTest, TlsHandshakeFailure_NoSharedCipher) {
+  setupTlsInspectorFilter(ConfigHelper::ServerSslOptions{}.setEcdsaCert(false).setRsaCert(true));
+  initialize();
+
+  Ssl::ClientSslTransportOptions ssl_options;
+  ssl_options.setClientEcdsaCert(true);
+  ssl_options.setCipherSuites({"ECDHE-ECDSA-AES128-GCM-SHA256"});
+  ssl_options.setTlsVersion(envoy::extensions::transport_sockets::tls::v3::TlsParameters::TLSv1_2);
+  setupSslConnection(/*expect_connection_open=*/false,
+                     /*ssl_options=*/ssl_options, /*curves_list=*/"P-256");
+
+  ASSERT_TRUE(waitForAccessLogConnection());
+  ASSERT_TRUE(waitForAccessLogStream());
+  ASSERT_TRUE(
+      // note: no peer certificate properties here, because the handshake fails
+      // at an earlier stage
+      waitForAccessLogRequest(fmt::format(R"EOF(
```

Comment:
I suspect this is going to be fragile with any access logs changes.  consider looking for specific strings instead

Reply:
This is how the other tests in this file worked, though I do agree that it's a bit unsightly. The existing test fixture is only set up to match on entire messages. What would you suggest?